src/error.rs

```diff
@@ -9,17 +9,38 @@ use std::str;
 
 use {raw, Session};
 
+/// An error code originating from a particular source.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum ErrorCode {
+    /// Codes for errors that originate in libssh2.
+    /// Can be one of  `LIBSSH2_ERROR_*` constants.
+    Session(libc::c_int),
+
+    /// Codes for errors that originate in the SFTP subsystem.
+    /// Can be one of `LIBSSH2_FX_*` constants.
+    //
+    // TODO: This should be `c_ulong` instead of `c_int` because these constants
+    // are only returned by `libssh2_sftp_last_error()` which returns `c_ulong`.
+    SFTP(libc::c_int),
+}
+
+impl fmt::Display for ErrorCode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
 /// Representation of an error that can occur within libssh2
 #[derive(Debug)]
 #[allow(missing_copy_implementations)]
 pub struct Error {
-    code: libc::c_int,
+    code: ErrorCode,
     msg: Cow<'static, str>,
 }
 
 impl Error {
     #[doc(hidden)]
-    pub fn last_error_raw(raw: *mut raw::LIBSSH2_SESSION) -> Option<Error> {
+    pub fn last_session_error_raw(raw: *mut raw::LIBSSH2_SESSION) -> Option<Error> {
```

Comment:
even though this is hidden, this is a breaking change to the Error api.  I think there's an async crate that may depend upon this.  Probably safest to bump ssh2 to `0.9.0`.

Reply:
This makes sense. It would also make sense with the bump of the sys crate. I'll fix this PR and prepare the next one.